website/identifiers/tasks.py

```diff
@@ -17,3 +17,32 @@ def task__update_doi_metadata_on_change(self, target_guid):
 @celery_app.task(ignore_results=True)
 def update_doi_metadata_on_change(target_guid):
     task__update_doi_metadata_on_change(target_guid)
+
+@celery_app.task(bind=True, max_retries=5, acks_late=True)
+def task__update_doi_metadata_with_verified_links(self, target_guid, verified_links=None, link_resource_type='Text'):
+    sentry.log_message('Updating DOI with verified links for guid',
+                      extra_data={'guid': target_guid, 'verified_links': verified_links, 'link_resource_type': link_resource_type},
+                      level=logging.INFO)
+
+    Guid = apps.get_model('osf.Guid')
+    target_object = Guid.load(target_guid).referent
+    try:
+        if verified_links is not None:
+            target_object.verified_links = verified_links
+            target_object.link_resource_type = link_resource_type
```

Comment:
Verified link and link resource type are not in OSF models. I thought we can insert them into the Datacite metadata for a project without storing them on the node. However, after seeing your code, I think we may have to do it this way. Please follow up with @opaduchak on this. A few observations here:

* Looks like each Datacite metadata update for a project/node rebuild the metadata from scratch. In this case, we need GV to send all configured links and their respective resource types to the task. So the task should receive a list of link-type pairs.
* If we need to store the links and resource types on the node, we need to update the OSF models. However, having duplicate data in both GV and OSF is a concern and may need more investigation on how we want to do this.